src/Mailer/AbstractTransport.php

```diff
@@ -34,10 +34,10 @@ abstract class AbstractTransport
     /**
      * Send mail
      *
-     * @param \Cake\Mailer\Email $email Email instance.
+     * @param \Cake\Mailer\Message $message Email mesage.
      * @return array
      */
-    abstract public function send(Email $email): array;
+    abstract public function send(Message $message): array;
```

Comment:
Won't this be a breaking change that we didn't have a deprecation for in 3.x?

This could impact any plugins or application transport implementations.

Reply:
Yes custom transports would be affected.

A compromise would be removing the typehint and allowing method to accept both `Email` and `Message` instances.

Reply:
Actually apart for adding the type hint no change will be required in transport implementations of plugins and apps and `Message` has the same public methods as `Email` to get headers and message.

Add typehints is something that's going to be required in general for all classes updating from 3.x to 4.0.

Reply:
Good point. Other transports will need to add *a* typehint either way. This change just changes the class name that ends up in the hint.